src/rct1/S4Importer.cpp

```diff
@@ -890,6 +889,7 @@ void S4Importer::ImportResearch()
     // gResearchExpectedDay =
     // gResearchExpectedMonth =
 
+	sub_684AC3();
```

Comment:
This function resets the progress and stage etc. This needs to be moved up to just after ```research_remove_non_separate_vehicle_types```